airflow/config_templates/config.yml

```diff
@@ -116,6 +116,17 @@ core:
       type: string
       example: ~
       default: "16"
+    max_consecutive_failed_dag_runs_per_dag:
+      description: |
+        The maximum number of consecutive DAG failures before DAG is automatically paused.
+        This is also configurable per DAG level with ``max_consecutive_failed_dag_runs``,
+        which is defaulted as ``max_consecutive_failed_dag_runs_per_dag``.
+        If not specified, then the value is considered as None,
+        meaning that the dags are never paused out by default.
```

Comment:
Lets mention this as experimental feature.

We need also:
1. News fragment entry for feature (that gives 2-3 sentences about this capability and explain this is experimental one)
2. Doc entry that explain this feature and explain this is experimental one


After these LGTM